src/System.Net.Http/src/System/Net/Http/HttpUtilities.cs

```diff
@@ -29,10 +29,16 @@ internal static bool IsHttpUri(Uri uri)
             IsSupportedSecureScheme(scheme);
 
         internal static bool IsSupportedNonSecureScheme(string scheme) =>
-            string.Equals(scheme, "http", StringComparison.OrdinalIgnoreCase);
+            string.Equals(scheme, "http", StringComparison.OrdinalIgnoreCase) || IsNonSecureWebSocketScheme(scheme);
 
         internal static bool IsSupportedSecureScheme(string scheme) =>
-            string.Equals(scheme, "https", StringComparison.OrdinalIgnoreCase);
+            string.Equals(scheme, "https", StringComparison.OrdinalIgnoreCase) || IsSecureWebSocketScheme(scheme);
```

Comment:
Note: In this code block, we allow ws and wss scheme.

Reply:
@caesar1995, can you elaborate?  It looks like it's checking for https and wss, not ws and wss...?

Reply:
The entire code block is from line 27 to 35.
https://github.com/caesar1995/corefx/blob/5848562b74862e7fa0a3e6987e40e2863507fe2c/src/System.Net.Http/src/System/Net/Http/HttpUtilities.cs#L27-L35

`IsSupportedNonSecureScheme` checks if http/ws.

`IsSupportedSecureScheme` checks if https/wss.

And `IsSupportedScheme` checks if `IsSupportedNonSecureScheme` || `IsSupportedSecureScheme`. 

When we create a `HttpRequestMessage` object, we will do `IsSupportedScheme` check on the uri. (If not meet, throws `ArgumentException`).

Reply:
> The entire code block is from line 27 to 35

Oh, sure, I thought you were just referring to the particular line.